fabric-block-view-api-v2/src/main/java/net/fabricmc/fabric/api/blockview/v2/FabricBlockView.java

```diff
@@ -0,0 +1,98 @@
+	 * where the user knows that the current context may be multithreaded, such as chunk building, to
+	 * ensure thread safety and data consistency. Using a {@link BlockEntity} directly may not be
+	 * thread-safe since it may lead to non-atomic modification of the internal state of the
+	 * {@link BlockEntity} (such as through lazy computation). Using a {@link BlockEntity} directly
+	 * may not be consistent since the internal state of the {@link BlockEntity} may change on a
+	 * different thread.
+	 *
+	 * <p>As previously stated, a common environment to use this method in is chunk building. Methods
+	 * that are invoked during chunk building and that thus should use this method include, but are
+	 * not limited to, {@code FabricBakedModel#emitBlockQuads} (block models),
+	 * {@code BlockColorProvider#getColor} (block color providers), and
+	 * {@code FabricBlock#getAppearance} (block appearance computation).
+	 *
+	 * <p>Users of this method are required to  check the returned object before using it.Users must
+	 * check if it is null and if it is of the correct type to avoid null pointer and class cast
+	 * exceptions, as the returned data is not guaranteed to be what the user expects. A simple way
+	 * to implement these checks is to use {@code instanceof}, since it always returns {@code false}
+	 * if the object is null. If the {@code instanceof} returns {@code false}, a fallback path should
+	 * be used.
+	 *
+	 * @param pos the position of the block entity
+	 * @return the render data provided by the block entity, or null if there is no block entity at this position
+	 *
+	 * @see RenderDataBlockEntity
+	 */
+	@Nullable
+	default Object getBlockEntityRenderData(BlockPos pos) {
+		BlockEntity blockEntity = ((BlockView) this).getBlockEntity(pos);
+		return blockEntity == null ? null : blockEntity.getRenderData();
+	}
+
+	/**
+	 * Checks whether biome retrieval is supported. The returned value will not change between
+	 * multiple calls of this method. See {@link #getBiomeFabric(BlockPos)} for more information.
+	 *
+	 * @return whether biome retrieval is supported
+	 * @see #getBiomeFabric(BlockPos)
+	 */
+	default boolean hasBiomes() {
+		return false;
+	}
+
+	/**
+	 * Retrieves the biome at the given position if biome retrieval is supported. If
+	 * {@link #hasBiomes()} returns {@code true}, this method will always return a non-null
+	 * {@link RegistryEntry} whose {@link RegistryEntry#value() value} is non-null. If
+	 * {@link #hasBiomes()} returns {@code false}, this method will always return {@code null}.
+	 *
+	 * @implNote Implementations which do not return null are encouraged to use the plains biome as
+	 * the default value, for example when the biome at the given position is unknown.
+	 *
+	 * @param pos the position for which to retrieve the biome
+	 * @return the biome, or null if biome retrieval is not supported
+	 * @see #hasBiomes()
+	 */
+	default RegistryEntry<Biome> getBiomeFabric(BlockPos pos) {
```

Comment:
I am not sure of the name, but I don't have a better suggestion. Should we also explain the difference with `WorldView#getBiome`? (i.e. that it is supported in more contexts)

Reply:
A section about `WorldView#getBiome` has been added to the javadoc in the latest commit.